server/protocol.c

```diff
@@ -1531,18 +1410,7 @@ request_rec *ap_read_request(conn_rec *conn)
     apply_server_config(r);
 
     if (!r->assbackwards) {
-        const char *tenc, *clen;
-
-        ap_get_mime_headers_core(r, tmp_bb);
-        apr_brigade_cleanup(tmp_bb);
-        if (r->status != HTTP_OK) {
-            ap_log_rerror(APLOG_MARK, APLOG_DEBUG, 0, r, APLOGNO(00567)
-                          "request failed: error reading the headers");
-            access_status = r->status;
-            goto die_unusable_input;
-        }
-
-        clen = apr_table_get(r->headers_in, "Content-Length");
+        const char *clen = apr_table_get(r->headers_in, "Content-Length");
```

Comment:
IIUC, we should have accurate r->body_indeterminate at this point, so check !r->body_indeterminate here and fail/internal error if no C-L in this case?

Reply:
I believe this is too early. Currently, we set this in the post_read hook. Also, requests may have neither C-L or body_indeterminate, e.g. if they are known to have no request body.

Maybe converting body_indeterminate into a no_body flag would make things more clear?